### ISSUE TITLE
Split nightly builds between test and compilation stages

### ISSUE THREAD

**Issue by hultonha** [o3de/o3de#12689]
**Is your feature request related to a problem? Please describe.**

It is currently not easy to tell if the nightly builds are failing because of a build/compilation failure or a failing test.

**Describe the solution you'd like**

Right now the nightly builds have both test runs and platform compilation all grouped into one. The tests routinely fail and haven't been green for a long time but the platform compilation step is usually green and people do from time to time inadvertently break it. To make the compilation breaks easier to spot, we suggest splitting out the the Jenkins pipeline for nightly builds to report two separate results. One for compilation (which should stay green) and one for tests (which is okay to remain red in the short-term but we would also like to get this to green eventually).

**Describe alternatives you've considered**

- Getting tests to green (time consuming, slow)
- Improving/changing output (visually separate compilation/test pass/failure state)

**Additional context**

For more context if required please reach out to @hultonha and/or @moraaar for details. We'll do our best to attend the next sig-build triage meeting as well.

Also tagging @dshmz for visibility as well.

**Comment by moraaar** [comment]
It'd be great to get this separation in both **o3de nightly** (https://jenkins.build.o3de.org/job/O3DE_periodic-incremental-daily/job/development/) and **aws-lumberyard-dev fork nightly** (https://jenkins-pipeline.agscollab.com/job/O3DE-LY-Fork-development_periodic-incremental-daily-internal/).

**Comment by amzn-changml** [comment]
As discussed in the SIG-Build meeting: Let's get some consensus on this, specifically with the SIGs involved with maintaining these tests. Anyone at @o3de/sig-testing and @o3de/sig-core want to chime in on this? 

For SIG-Build, we don't see a specific issue with this, but we do want to know if we want to run this at a differing cadence for cost reasons. Not all jobs may need to run daily, and can run at slower intervals if there's longer running tests that could take more than the window of time allocated.

**Comment by hultonha** [comment]
> As discussed in the SIG-Build meeting: Let's get some consensus on this, specifically with the SIGs involved with maintaining these tests. Anyone at @o3de/sig-testing and @o3de/sig-core want to chime in on this?
> 
> For SIG-Build, we don't see a specific issue with this, but we do want to know if we want to run this at a differing cadence for cost reasons. Not all jobs may need to run daily, and can run at slower intervals if there's longer running tests that could take more than the window of time allocated.

Thank you very much for following-up @amzn-changml. To try and increase the chance @o3de/sig-testing and @o3de/sig-core see this and respond, I'm going to add the sigs to the issue and also re-add the `needs-triage` label to hopefully get a response in their triage sessions.

**Comment by hultonha** [comment]
Requesting feedback/input from @o3de/sig-core and @o3de/sig-testing. Thank you!

**Comment by lumberyard-employee-dm** [comment]
For SIG-Core, the split of the nightly build pipeline to a higher priority daily job phase and a lower priority weekly job phase would help with address the stability of the nightly job in general.

A benefit for splitting the o3de-nightly into pipeline 2 which has all the periodic, sandbox and benchmark jobs is that we can provide a higher timeout mechanism for the those test, which would allow them to complete.
Currently the Linux benchmark jobs timeout for the AZ Core benchmarks after it reaches the 1500 second limit.
Splitting the pipeline would allow us to configure the timeout separately from the smoke and main jobs.

For the aws-lumberyard-dev fork pipeline split, I would actually suggest a 3rd pipeline for Mac/iOS jobs.
Mac and iOS take several hours to build aren't being actively maintained now. 
Also with the amount of Mac/iOS jobs proposed I believe that the nightly job would get backed up so much that it would take up to a day to complete that part of the nightly build. 
Mac and iOS are currently building on physical hardware, which is limited and we can't run all their jobs in parallel.

Therefore I would suggest the Mac/iOS jobs be split into a separate pipeline from the Windows, Linux and Android jobs.
I also recommend pairing down the number of build configurations for the Mac and iOS jobs as well.
Not all build configurations of debug, profile, asset_profile, profile_nounity, release and monolithic_release are needed for those platforms and we can get good coverage by targeting specific combination of jobs.
Mac is not a target for a game/simulation release, it's main purpose is for development of an iOS game.
So the configurations I think we should target for Mac should be
* debug, asset_profile

For iOS the target is a game and specifically. IN that case we can target the configurations of
* monolithic_debug, profile, monolithic_release

Furthermore the iOS asset_profile job is really a Mac job that enables the asset processing of ios assets.
I think the Mac and iOS asset_profile jobs should be combined into a single job.
The Asset Processor can process both Mac and iOS assets in a single run and therefore multiple jobs aren't needed.

**Comment by moraaar** [comment]
Agree completely with @lumberyard-employee-dm. My only addition would be to also build "profile" for mac, which I don't see it in his list (unless it's expected to be part of asset_profile, which in that case that's covered already).

**Comment by lumberyard-employee-dm** [comment]
> Agree completely with @lumberyard-employee-dm. My only addition would be to also build "profile" for mac, which I don't see it in his list (unless it's expected to be part of asset_profile, which in that case that's covered already).

asset_profile builds both the profile configuration and runs the AssetProcessor

**Comment by hultonha** [comment]
Thank you very much for the feedback @lumberyard-employee-dm, super helpful!

**Comment by amzn-changml** [comment]
Thanks for the data points, I'll start scoping out the work

**Comment by Kadino** [comment]
Clarifying the stages seems useful. No other concerns from SIG-Testing.

**Comment by moraaar** [comment]
Any updates on this? @dshmz @amzn-changml

**Comment by amzn-changml** [comment]
> Any updates on this? @dshmz @amzn-changml

Starting the work as we've just completed the AR changes to support project based builds last week. It's on the roadmap and we're discussing it with TSC as well.
